test-project/PlaybackViewController.m

```diff
@@ -39,11 +40,34 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
   }
 }
 
+- (void) toggleFullscreen:(DMPlayerViewController *)player {
+
+    if (player.fullscreen) {
+        self.playerSize = self.initialPlayerSize;
+    }
+    else {
+        self.playerSize = self.view.frame.size;
+    }
+
+    self.playerWidthLayoutConstraint.constant = self.playerSize.width;
+    self.playerHeightLayoutConstraint.constant = self.playerSize.height;
+
+    // Once the transition from/to fullscreen is complete, notify the player so it can update its UI.
+    [player notifyFullscreenChange];
+}
+
 #pragma mark DMPlayerDelegate
 - (void)dailymotionPlayer:(DMPlayerViewController *)player didReceiveEvent:(NSString *)eventName {
   // Grab the "apiready" event to trigger an autoplay
   if ([eventName isEqualToString:@"apiready"]) {
-    [player play];
+    // From here, it's possible to interact with the player API.
+    NSLog(@"Received apiready event");
+  }
+
+  // When "fullscreen-action" parameter is set to "trigger_event", the transition from/to fullscreen must be handled by the host application
+  // This allow the host application to keep the player UI when going to fullscreen instead of depending on the native iOS player UI.
+  if ([[self.additionalParameters valueForKey:@"fullscreen-action"] isEqualToString:@"trigger_event"] && [eventName isEqualToString:@"fullscreen_toggle_requested"]) {
```

Comment:
"modern" ObjC syntax for dictionary access would be preferable: `self.additionalParameters[@"fullscreen-action"]`